### PR TITLE
1.0.0

### PR DESCRIPTION
**Pull request by andrasguseo**

### CODE REVIEW

**Comment by andrasguseo** [comment]
This is the ticket number:
https://central.tri.be/issues/91475

The extension is a 1:1 implementation of this snippet:
https://gist.github.com/andrasguseo/b8e2739b11fdbd2d5ebacce23d862046

File: index.php [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+			if ( ! isset( $_GET['ical'] )
+			     || ! isset( $query->tribe_is_event_query )
+			     || ! $query->tribe_is_event_query ) {
+				return;
+			}
+
+			$tribe_display	= $_GET[ 'tribe_display' ];
+			$start_date	    = isset( $_GET[ 'start_date' ] ) ? $_GET[ 'start_date' ] : "";
+			$end_date	    = isset( $_GET[ 'end_date' ] )   ? $_GET[ 'end_date' ]   : "";
+
```
Comment (nefeline) [inline index.php]: I would recommend fixing the format here (align the equal signs and remove unnecessary spaces). Also, I would recommend replacing the double quotes with single quotes here.

File: index.php [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+			// Insert filters and hooks here
+			add_action( 'pre_get_posts', array( $this, 'filter_ical_query' ) );
+		}
+
+		/**
+		 * Filtering the query for dates
+		 *
+		 * @param $query
+		 */
+		function filter_ical_query( $query ) {
```
Comment (nefeline) [inline index.php]: The visibility must be declared on this method: `public function filter_ical_query( $query ) {`

File: index.php [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+		/**
+		 * Validates the date
+		 *
+		 * param $date
+		 * @param string $format
+		 *
+		 * @return bool
+		 */
+		function validateDate( $date, $format = 'Y-m-d H:i:s' ) {
```
Comment (nefeline) [inline index.php]: The visibility must be declared for this method too. Also, update the name to match the same pattern found in other methods (snake_case): `public function validate_date( $date, $format = 'Y-m-d H:i:s' ) {`

File: index.php [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU General Public License for more details.
+ */
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if (
+	class_exists( 'Tribe__Extension' )
+	&& ! class_exists( 'Tribe__Extension__Advanced_iCal_Export' )
+) {
```
Comment (nefeline) [inline index.php]: I would recommend formatting this in a single line:
`if ( class_exists( 'Tribe__Extension' ) && ! class_exists( 'Tribe__Extension__Export_Events_Modern_Events_Calendar' ) ) {
`

File: index.php [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+			 * Protect against fatals by specifying the required minimum PHP
+			 * version. Make sure to match the readme.txt header.
+			 */
+			$php_required_version = '5.6';
+
+			if ( version_compare( PHP_VERSION, $php_required_version, '<' ) ) {
+				if (
+					is_admin()
+					&& current_user_can( 'activate_plugins' )
+				) {
```
Comment (nefeline) [inline index.php]: Let's modify this conditional to a single line too: 
`if ( is_admin() && current_user_can( 'activate_plugins' ) ) {`

File: index.php [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+			     || ! isset( $query->tribe_is_event_query )
+			     || ! $query->tribe_is_event_query ) {
+				return;
+			}
+
+			$tribe_display	= $_GET[ 'tribe_display' ];
+			$start_date	    = isset( $_GET[ 'start_date' ] ) ? $_GET[ 'start_date' ] : "";
+			$end_date	    = isset( $_GET[ 'end_date' ] )   ? $_GET[ 'end_date' ]   : "";
+
+			if( $tribe_display === 'custom' ) {
```
Comment (nefeline) [inline index.php]: Put spaces on both sides of the opening and closing parentheses (WP code standards):  `if ( $tribe_display === 'custom' ) {`
Same for lines 97, 117, 119, 123 and 128.

File: index.php [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+				// Check if there is a start_date set
+				if( isset( $start_date ) && !empty( $start_date ) ) {
```
Comment (nefeline) [inline index.php]: Same as mentioned before: add a space between if and the opening parenthesis here: `if ( isset( $start_date ) && !empty( $start_date ) ) {`

File: index.php [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+		 *
+		 * @return bool
+		 */
+		function validateDate( $date, $format = 'Y-m-d H:i:s' ) {
+			$d = DateTime::createFromFormat( $format, $date );
+			return $d && $d->format( $format ) == $date;
+		}
+
+	} // end class
+} // end if class_exists check
```
Comment (nefeline) [inline index.php:150]: Good job here @andrasguseo :beers:!

Comment (andrasguseo) [reply index.php:150]: I stole this part. :)

**Comment by andrasguseo** [comment]
@nefeline : addressed your points. Please re-check. :)

**Comment by nefeline** [comment]
Merge away! :dancer: :wine_glass:

**Pull request merged**